cpu/samd21/cpu.c

```diff
@@ -118,25 +156,14 @@ static void clk_init(void)
 
     /* setup generic clock 1 as 1MHz for timer.c */
     GCLK->GENDIV.reg = (GCLK_GENDIV_DIV(8) |
-                        GCLK_GENDIV_ID(1));
+                        GCLK_GENDIV_ID(SAM0_GCLK_1MHZ));
     GCLK->GENCTRL.reg = (GCLK_GENCTRL_GENEN |
                          GCLK_GENCTRL_SRC_OSC8M |
-                         GCLK_GENCTRL_ID(1));
-    while (GCLK->STATUS.reg & GCLK_STATUS_SYNCBUSY) {}
-
-    /* Setup clock GCLK3 with divider 1 */
-    GCLK->GENDIV.reg = GCLK_GENDIV_ID(3) | GCLK_GENDIV_DIV(1);
-    while (GCLK->STATUS.reg & GCLK_STATUS_SYNCBUSY) {}
-
-    /* Enable GCLK3 with XOSC32K as source */
-    GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(3) |
-                        GCLK_GENCTRL_GENEN |
-                        GCLK_GENCTRL_RUNSTDBY |
-                        GCLK_GENCTRL_SRC_XOSC32K;
+                         GCLK_GENCTRL_ID(SAM0_GCLK_1MHZ));
     while (GCLK->STATUS.reg & GCLK_STATUS_SYNCBUSY) {}
 
-    /* set GCLK3 as source for DFLL */
-    GCLK->CLKCTRL.reg = (GCLK_CLKCTRL_GEN_GCLK3 |
+    /* set GCLK2 as source for DFLL */
+    GCLK->CLKCTRL.reg = (GCLK_GENDIV_ID(SAM0_GCLK_32KHZ) |
```

Comment:
Oops, that should be `GCLK_CLKCTRL_GEN()`

Reply:
Thank you!